### PR TITLE
Integrate chart into Stats

### PR DESCRIPTION
**Pull request by EugeneTeu**
Screenshot 2019-10-19 at 11 56 46 PM

### CODE REVIEW

File: src/main/java/seedu/address/statistic/StatisticManager.java [hunk 1]
```diff
@@ -23,6 +29,84 @@
...
+        // loops through the list of months, for each month, calculate the revenue for all orders in that month
+        // returns this in the format of XYChat.DaTa().
+        List<XYChart.Data<String, Number>> listOfMonthlyProfit =
+                listOfMonth.stream()
+                        .map(month -> processProfitByMonth(listOfFilteredOrders, month))
+                        .collect(Collectors.toList());
+
+        listOfMonthlyProfit.stream().forEach(x -> series.getData().add(x));
+        //*****to remove when merging*********
+        //series.getData().forEach(x -> System.out.println(x));
```
Comment (uberSaiyan) [inline src/main/java/seedu/address/statistic/StatisticManager.java]: Is this supposed to removed?

File: src/main/java/seedu/address/statistic/DateUtil.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public static List<Calendar> getListOfYearMonth(StatsPayload statsPayload) {
+        List<Calendar> listOfYearMonth = new ArrayList<>();
+        Calendar startDate = statsPayload.getStartingDate();
+        Calendar endDate = statsPayload.getEndingDate();
+
+
+        while (startDate.before(endDate)) {
+            //System.out.println(convertCalendarDateToString(startDate));
```
Comment (uberSaiyan) [inline src/main/java/seedu/address/statistic/DateUtil.java]: Is this supposed to removed?

File: src/main/java/seedu/address/statistic/StatisticManager.java [hunk 3]
```diff
@@ -46,7 +130,86 @@ public String calculateTotalCostOnCompleted(ReadOnlyDataBook<Order> orderBook, S
...
+    /**
+     * Takes in an orderlist, calculates the profit of all orders in this month
+     */
+    private static double calculateProfitByMonth(List<Order> orderList, Calendar month) {
+        double[] doubleProfitList =
+                StatisticManager.checkIfOrderIsPresent(orderList.stream())
+                        .filter(currentOrder ->
+                                DateUtil.extractMonth(currentOrder) == month.get(2)
+                                        && DateUtil.extractYear(currentOrder) == month.get(1))
+                        .map(currentOrder ->
```
Comment (uberSaiyan) [inline src/main/java/seedu/address/statistic/StatisticManager.java]: Perhaps can directly mapToDouble here?

File: src/main/java/seedu/address/ui/StatisticsWindow.java [hunk 4]
```diff
@@ -10,18 +16,42 @@
...
-    public StatisticsWindow(Stage root) {
+    public StatisticsWindow(Stage root, Optional<XYChart.Series<String, Number>> axisSeriesOptional) {
         super(FXML, root);
+        if (axisSeriesOptional.isPresent()) {
+            this.axisSeries = axisSeriesOptional.get();
+            buildChart();
+        } else {
+
+        }
+
```
Comment (uberSaiyan) [inline src/main/java/seedu/address/ui/StatisticsWindow.java]: Is there supposed to be more code here?

File: src/main/java/seedu/address/statistic/StatisticManager.java (hunk 1 above)
Comment (EugeneTeu) [reply src/main/java/seedu/address/statistic/StatisticManager.java]: ye

File: src/main/java/seedu/address/statistic/DateUtil.java (hunk 2 above)
Comment (EugeneTeu) [reply src/main/java/seedu/address/statistic/DateUtil.java]: ye

File: src/main/java/seedu/address/ui/StatisticsWindow.java (hunk 4 above)
Comment (EugeneTeu) [reply src/main/java/seedu/address/ui/StatisticsWindow.java]: no LOL

File: src/main/java/seedu/address/statistic/StatisticManager.java (hunk 3 above)
Comment (EugeneTeu) [reply src/main/java/seedu/address/statistic/StatisticManager.java]: You mean I convert then map? cause I already use the Moneyutil to convert before I subtract

**Pull request merged**